src/app/shared/provider.service.ts

```diff
@@ -24,9 +27,30 @@ export class ProviderService {
 
     tool.provider = this.getProvider(gitUrl);
     tool.providerUrl = this.getProviderUrl(gitUrl, tool.provider);
+    tool.providerIcon = this.getProviderIcon(tool.providerUrl);
     return tool;
   }
 
+  private getProviderIcon(providerUrl: string): IconDefinition {
+    if (!providerUrl) {
+      return null;
+    }
+    if (providerUrl.startsWith('https://github.com/')) {
```

Comment:
There is a method right below that already determines the registry based on the gitUrl. Instead of having your own logic to determine the registry, I suggest using that method, something like:

```
const provider = getProvider(gitUrl)
if (provider === 'GitHub') return faGitHub
etc.
```

Reply:
Implemented using the said method.